discord_bots/cogs/map.py

```diff
@@ -1,172 +1,237 @@
-from discord.ext.commands import Bot, Context, check, command
+import logging
+from table2ascii import Alignment, PresetStyle, table2ascii
+from typing import List, Optional
+from sqlalchemy.orm.session import Session as SQLAlchemySession
+
+from discord import (
+    app_commands,
+    Colour,
+    Embed,
+    Interaction,
+)
+
+from discord.ext.commands import Bot
 from sqlalchemy.exc import IntegrityError
 from sqlalchemy.orm.exc import NoResultFound
 
-from discord_bots.checks import is_admin
+from discord_bots.checks import is_admin_app_command, is_command_channel
 from discord_bots.cogs.base import BaseCog
 from discord_bots.models import (
+    Category,
     FinishedGame,
+    FinishedGamePlayer,
     InProgressGame,
     Map,
     MapVote,
+    PlayerCategoryTrueskill,
     Queue,
     Rotation,
     RotationMap,
+    Session,
 )
-from discord_bots.utils import send_message
+from discord_bots.utils import code_block, short_uuid, win_rate
+
+_log = logging.getLogger(__name__)
 
 
 class MapCommands(BaseCog):
     def __init__(self, bot: Bot):
         super().__init__(bot)
 
-    @command()
-    @check(is_admin)
-    async def addmap(self, ctx: Context, map_full_name: str, map_short_name: str):
+    group = app_commands.Group(name="map", description="Map commands")
+
+    @group.command(name="add", description="Add a map to the map pool")
+    @app_commands.check(is_admin_app_command)
+    @app_commands.check(is_command_channel)
+    @app_commands.describe(full_name="Long name of map", short_name="Short name of map")
```

Comment:
Instead of outright renaming "map_short_name" to "map_name", you could have used the app_commans.rename decorator. This was probably written before you knew about it or something, but the refactor is fine.